test/unit/processing/validate-sequence.test.js

```diff
@@ -0,0 +1,123 @@
+let batch
+let validateSequence
+const { sfi, sfiPilot, lumpSums } = require('../../../app/schemes')
+
+const setupMocks = (mockDisableSequenceValidation = false) => {
+  jest.mock('../../../app/processing/batch')
+  batch = require('../../../app/processing/batch')
+  jest.mock('../../../app/config/processing', () => ({
+    disableSequenceValidation: mockDisableSequenceValidation
+  }))
+  validateSequence = require('../../../app/processing/validate-sequence')
+}
+
+describe('Validate sequence', () => {
+  beforeEach(() => {
+    jest.resetModules()
+  })
+
+  test('returns success if next sequence matches expected for SFI', async () => {
```

Comment:
I'm not sure there's much value in those tests.  schemeId is a constant it couldn't be any of those.  If it was it would cause other higher level tests we've got to fail.  I guess it could be undefined so I've added one for that scenario.

Now the rollover is interesting.  You're absolutely correct that when we write files to DAX, we just start again at `1`.  DAX doesn't actually validate or read the sequence.  It's just to help distinguish the different files.

But with inbound files from Siti Agri, I don't know what Siti Agri will do when it gets to `9999`.  For BPS in seven years it's only up to `4535` and for CS it's up to `3849`.  So it's not going to happen anytime soon.  But would good to know what it will do so we can handle it here.  Will it start again at `1`?  Will it extend the filename and go to `10000`?  Will it break?

I've created a ticket so we can explore and make whatever changes needed based on the answer.